### PR TITLE
Improve main method documentation

### PR DESCRIPTION
**Pull request by ehmicky**
Users now have the choice between 4 methods: `execa()`, `execaCommand()`, `execaNode()` and `$`. This PR attempts to simplify the documentation of each of them and focus on their difference and on why users should use one over the other.

It also consolidates their documentation. For example, escaping/quoting was mentioned in different places, with different wording. Each method now includes only the information relevant for it. A dedicated "Shell syntax" section explains the bit about that, since that's a common source of confusion.

A few sections could also be greatly simplified without losing any meaningful information, in my opinion. 

Overall, I'm hoping this makes the main entry points easier to understand for new Execa users.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -613,72 +615,26 @@ type TemplateExpression =
...
-	@throws A `childProcessResult` error
-
-	@example <caption>Basic</caption>
-	```
-	import {$} from 'execa';
-
-	const {stdout} = await $`echo unicorns`;
-	// const {stdout} = await $`echo ${'unicorns'}`;
-	// const {stdout} = await $`echo ${['unicorns', 'rainbows']}`;
+	Returns a new instance of `$` but with different default `options`. Consecutive calls are merged to previous ones.
```
Comment (ehmicky) [inline index.d.ts:618]: Those code comments were accidentally documenting `` $`command` `` instead of `$(options)`.

File: readme.md [hunk 2]
```diff
@@ -300,24 +330,6 @@ Combines both [`pipeStdout()`](#pipestdouttarget) and [`pipeStderr()`](#pipestde
 
 Either the [`stdout` option](#stdout-1) or the [`stderr` option](#stderr-1) must be kept as `pipe`, their default value. Also, the [`all` option](#all-2) must be set to `true`.
 
-### execaSync(file, arguments?, options?)
```
Comment (ehmicky) [inline readme.md:303]: Those lines were not put in the right place.

**Comment by sindresorhus** [comment]
Looks good

**Comment by sindresorhus** [comment]
Can you fix the merge conflict?

**Comment by ehmicky** [comment]
Fixed :+1:

**Comment by ehmicky** [comment]
Fixed :+1: 
Should I merge?

**Pull request merged**